ReactiveCocoa/Swift/Signal.swift

```diff
@@ -299,24 +299,168 @@ extension SignalType {
 	}
 }
 
-/// A reference type which wraps an array to avoid copying it for performance and
-/// memory usage optimization.
+/// A reference type which wraps an array to auxiliate the collection of values
+/// for `collect` operator.
 private final class CollectState<Value> {
 	var values: [Value] = []
 
-	func append(value: Value) -> Self {
+	/// Collects a new value.
+	func append(value: Value) {
 		values.append(value)
-		return self
+	}
+
+	/// Check if there are any items remaining.
+	///
+	/// - Note: This method also checks if there weren't collected any values 
+	/// and, in that case, it means an empty array should be sent as the result
+	/// of collect.
+	var isEmpty: Bool {
+		/// We use capacity being zero to determine if we haven't collected any 
+		/// value since we're keeping the capacity of the array to avoid 
+		/// unnecessary and expensive allocations). This also guarantees 
+		/// retro-compatibility around the original `collect()` operator.
+		return values.isEmpty && values.capacity > 0
+	}
+
+	/// Removes all values previously collected if any.
+	func flush() {
+		// Minor optimization to avoid consecutive allocations. Can
+		// be useful for sequences of regular or similar size and to
+		// track if any value was ever collected.
+		values.removeAll(keepCapacity: true)
 	}
 }
 
+
 extension SignalType {
-	/// Returns a signal that will yield an array of values when `self` completes.
+
+	/// Returns a signal that will yield an array of values until it reaches a 
+	/// certain count.
+	///
+	/// When the count is reached the array is sent and the signal starts over 
+	/// yielding a new array of values.
+	///
+	/// - Precondition: `count` should be greater than zero.
+	///
```

Comment:
Can you also note that any remaining values will be sent when the signal completes? (i.e. the last array may not have `count` items.)